### PR TITLE
[DBX-67497] Add date validity check for separation date

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- Add validity check input to user provided 526 date fields
- Note that we were unable to recreate the original 'impossible date' bug.  TL;DR 526 submissions were failing because have been sending EVSS impossible dates such as November 31st, June 29th, etc.  This solution will at best fix the problem, at worse do nothing.  It's a sensible check to have in place in any case.

### Unrelated to feature
- add `.ignore` to git ignore for vim search config

## Related issue(s)

- [Ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/67497)
- [Slack thread about priority of this work](https://dsva.slack.com/archives/C04KW0B46N5/p1697133617275569?thread_ts=1697050957.612989&cid=C04KW0B46N5)

## Testing done

- Because we were unable to recreate the original bug, we needed to manually insert a 'bad date' using the chrome dev tools or manual string manipulation.  Either way, once we have an impossible date in the date field, we see the error displayed as expected.

Screenshot 2023-10-23 at 10 04 13 AM

- Happy path was validated with click through tests

## What areas of the site does it impact?

- 526 disability compensation flow
- any date field using the `disability-benefits/all-claims/validations` date validation set

## Acceptance criteria
- Impossible dates are rejected
- possible dates are accepted as normal

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated (N/A)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/platform/forms-system/src/js/widgets/DateWidget.jsx [hunk 1]
```diff
@@ -83,7 +84,7 @@ export default class DateWidget extends React.Component {
     const { month, day, year } = this.state.value;
     let daysForSelectedMonth;
 
-    const monthYear = options.monthYear;
```
Comment (SamStuckey) [inline src/platform/forms-system/src/js/widgets/DateWidget.jsx:86]: the linter is preventing me from removing this change.  It is a good change, but it is also unrelated to my commit.  Not sure what to do but ignore it.

**Review by mchae-nava — APPROVED** [review]
LGTM!

**Review by micahaspyr — APPROVED** [review]

**Comment by SamStuckey** [comment]
Converted back to 'draft' to address spec failures

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**